python/ray/experimental/state/common.py

```diff
@@ -956,8 +963,21 @@ def get_or_create_actor_task_group(
                     # tree at that node.
                     return None
 
-                # TODO(aguo): Get actor name from actors state-api.
-                [actor_name, *rest] = creation_task["func_or_class_name"].split(".")
+                # TODO(rickyx)
+                # We are using repr name for grouping actors if exists,
+                # else use class name. We should be using some group_name in the future.
+                if actor is None:
+                    logger.debug(
+                        f"We are missing actor info for actor {actor_id}, "
+                        f"even though creation task exists: {creation_task}"
+                    )
+                    [actor_name, *rest] = creation_task["func_or_class_name"].split(".")
+                else:
+                    actor_name = (
+                        actor["repr_name"]
```

Comment:
Doesn't repr_name always exist? Looks like it is also non optional value in the protobuf. We should probably distinguish the default repr vs non default otherwise the caradinality will be very high since the default repr includes the memory addr

Reply:
It's not optional in `gcs.proto` - it will be default value of a string, which is an empty string. 

When no `__repr__` is set, it will just be an empty string, see the screenshots attached above, as well as the test: https://github.com/ray-project/ray/pull/33555/files#diff-3ca7ac77a894dcb8f6ec467e0f0f48eb76983e991f86e38e3c7b81aa11492697R203

Reply:
oh interesting... Looks like the default python class has a repr. Maybe we disable it intentionally?

Reply:
We checked it explicitly: https://github.com/ray-project/ray/blob/94253587fe2ed412dcfe00f6ad8e52f51292a687/python/ray/_raylet.pyx#L930-L931

Reply:
makes sense...